src/main/java/dog/pawbook/model/managedentity/UniqueEntityList.java

```diff
@@ -216,13 +216,15 @@ private static boolean referencedIdValid(Pair<Integer, Entity> focus, List<Pair<
                 }
                 assert dogs.size() == 1 : "There should only be exactly one matching dog";
                 Dog dog = dogs.get(0);
+                assert(dog.getOwnerId() != null) : "OwnerID should not return a null";
```

Comment:
this instance here is redundant as well, since it will cause the if below to trip otherwise.